### PR TITLE
Module support

### PR DESCRIPTION
**Pull request by Bromeon**
Adds support for `mod` and `static`.

Modules currently can contain these item types (which can now be top-level `Declaration` as well):
* `const`
* `static` (new)
* `fn`
* `struct`
* `enum`
* `union`
* `impl`

These item types can be inside `mod`, but are not parsed as top-level `Declaration`s themselves:
* `#![inner]` attributes
* `use` statements

Not supported:
* Declarative macros
* Traits (will follow in a later PR)
* `extern crate`
* possibly some other token combinations I forgot

### CODE REVIEW

**Review by PoignardAzur — APPROVED** [review]
Overall, LGTM. There are things I think should be improved, but none of them seem hard blockers. I leave it to you whether to include the improvements I mentioned in this PR, or in follow-ups.

If you prefer the latter, I recommend you create an issue for each problem I mentioned, then merge the PR.

File: src/parse.rs
Comment (Bromeon) [reply src/parse.rs:179]: I tend to miss a lot of "not everyday Rust" syntax, glad to have reviews for this!

Would it be OK if I tried added support after the `unsafe impl` PR is merged? I have the feeling I could reuse some of the code structure introduced there.

File: src/parse.rs
Comment (Bromeon) [reply src/parse.rs]: No, removed.

File: src/parse.rs
Comment (Bromeon) [reply src/parse.rs]: Absolutely agree, I had too narrow of a use case in mind.
The previous approach also didn't support `pub use`, which is fairly common.

Fixed.

**Pull request merged**
